src/module_1/module_1_meteo_api.py

```
@@ -24,60 +25,68 @@ def call_API(url):
 
-    dataset = dataset.dropna(axis=1, how='all')
-    dataset['Average'] = dataset.mean(axis=1, skipna=True, numeric_only=True)
-    dataset['Dispersion'] = dataset.var(axis=1, skipna=True, numeric_only=True)
-
-    dataset['time'] = pd.to_datetime(dataset['time'])
-    dataset['year'] = dataset['time'].dt.year
+    dataset["time"] = pd.to_datetime(dataset["time"])
+    dataset["year"] = dataset["time"].dt.year
 
     return dataset
-    
 
-def plot_data(data, yaxis):
 
-    yearly_average = data.groupby('year')['Average'].mean().reset_index()
-    yearly_variance = data.groupby('year')['Dispersion'].mean().reset_index()
+def plot_data(data, yaxis):
+    yearly_average = data.groupby("year")["Average"].mean().reset_index()
+    yearly_std = data.groupby("year")["Dispersion"].mean().reset_index()
 
     plt.figure(figsize=(10, 6))
-    sns.lineplot(data=yearly_average, x='year', y='Average', marker='o',
-                linestyle='-', label=yaxis)
-
-    plt.fill_between(yearly_variance['year'], 
-                    yearly_average['Average'] - yearly_variance['Dispersion'], 
-                    yearly_average['Average'] + yearly_variance['Dispersion'], 
-                    color='lightblue', alpha=0.5, label='Dispersion')
-
-    plt.xlabel('Year')
+    sns.lineplot(
+        data=yearly_average,
+        x="year",
+        y="Average",
+        marker="o",
+        linestyle="-",
+        label=yaxis,
+    )
+
+    plt.fill_between(
+        yearly_std["year"],
+        yearly_average["Average"] - yearly_std["Dispersion"],
+        yearly_average["Average"] + yearly_std["Dispersion"],
+        color="lightblue",
+        alpha=0.5,
+        label="Dispersion",
+    )
+
+    plt.xlabel("Year")
     plt.ylabel(yaxis)
     plt.title(yaxis)
     plt.legend()
     plt.grid(True)
 
     plt.show()
 
-def main():
 
+def main():
     json_file = get_data_meteo_api("Madrid")
-    df = pd.DataFrame(json_file['daily'])
+    df = pd.DataFrame(json_file["daily"])
```

This will break if `status_code != 200` since `get_data_meteo_api` would return `None`. If you want any API failure to be the end of the street you should raise the exception within `call_API` rather than returning `None`.